### PR TITLE
Adding a dark mode toggle

### PR DESCRIPTION
**Pull request by ChrisGlad**
Signed-off-by: Christopher Gladman <chris.gladman2004@gmail.com>
Change-Id: I0e0c282cec24d9bb7fbfc4131ba3875976cea5fd

* Resolves: # <!https://github.com/CollaboraOnline/online/issues/5054>
* Target version: master 

### Summary
This makes changes to the Menu and Notebook bars, adding a new 
"Dark mode" option. 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by pedropintosilva** [comment]
Thanks @ChrisGlad  I will review it but maybe better to do a bit of rewording, here are some tips:
- Start the commit subject (AKA title) with a capital letter
- Add commit description
- This is not LibreOffice
- The commit message should tell what the commit does, in imperative mood. If necessary, you can mention how the code used to work, too. But what actually is done by the commit should be in present tense. Not what you did. Not "Fixed hang in foo()" but "Fix hang in foo()"
- The first line is supposed to be a minimal summary. Do not end it with a period.
- If the change is very simple conceptually, that one line might be enough
- The rest of the commit message describes the change in greater detail. This is the body of this commit message. The body is written after the subject line with one blank line in between. The blank line is used by various tools (such as 'git log', 'git show', etc.) to differentiate subject from body. Further paragraphs are also separated by blank lines.
- Explain the problem being solved by this commit. More importantly, explain why these changes are being made, as opposed to how. The 'why' part is your responsibility, the 'how' part is code's responsibility.

**Comment by pedropintosilva** [comment]
@ChrisGlad thanks but that doesn't change the commit message as you see when you do `git log` locally or when you click in the commits tab here in the  web interface

**Comment by pedropintosilva** [comment]
- `git commit --amend`
- `git push origin name-of-your-branch  --force` in this case I guess `Request`

**Review by pedropintosilva — CHANGES_REQUESTED** [review]
Also it would be good if the toggle would get selected upon switching to dark mode

File: browser/src/control/Control.NotebookbarWriter.js [hunk 1]
```diff
@@ -1085,6 +1085,12 @@ L.Control.NotebookbarWriter = L.Control.Notebookbar.extend({
 				],
 				'vertical': 'true'
 			},
+			{
+				'id':'toggledarktheme',
+				'type': 'bigmenubartoolitem',
```
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarWriter.js:1090]: Cool, nice to use bigmenubartoolitem so we can share stugg between Menubar and notebookbar :+1:

**Comment by ChrisGlad** [comment]
> Also it would be good if the toggle would get selected upon switching to dark mod

The code should add a tick in the Menubar item when dark mode is active. Do you want me to try something similar for the Notebookbar?

**Comment by pedropintosilva** [comment]
> > Also it would be good if the toggle would get selected upon switching to dark mod
> 
> The code should add a tick in the Menubar item when dark mode is active. Do you want me to try something similar for the Notebookbar?

WRT menubar it looks nice! For the Notebookbar I meant that the would be good to have the button being set to `selected` when the dark mode is active (We have other buttons doing something similar such as ruler)

File: browser/src/control/Control.UIManager.js [hunk 2]
```diff
@@ -51,6 +51,38 @@ L.Control.UIManager = L.Control.extend({
 			|| forceCompact === false;
 	},
 
+	toggleDarkMode: function() {
+		// the initial theme is 'light' unless it is already set to 'dark'
+		var InitialScheme = 'light';
```
Comment (eszkadev) [inline browser/src/control/Control.UIManager.js:56]: we use camelCaseStartingFromSmallCharacter :)

**Review by lpranam — CHANGES_REQUESTED** [review]
3 problems I noticed.

1. in Tabbed view button does not show if selected
2. If i reopen the doc or refresh the page, theme is again set to light theme. Interesting thing is, if it was supposed to be dark mode, in compact view dark mode option is checked even if it is currently showing light mode. in short in compact view, menu option is correctly checked even if theme is displayed incorrectly.
3. dark mode icon seems a little pixelated and also moon has a light gray area on the bottom edge but I think this can be fixed as follow-up.

**Comment by pedropintosilva** [comment]
@lpranam 

> 3 problems I noticed.
> 
>     1. in Tabbed view button does not show if selected
I think I have pointed out this in https://github.com/CollaboraOnline/online/pull/5092#issuecomment-1202313115, but let's let that for a follow-up PR. Right now we really need to take out dark mode from being loaded automatically as it is without this PR.
> 
>     2. If i reopen the doc or refresh the page, theme is again set to light theme. Interesting thing is, if it was supposed to be dark mode, in compact view dark mode option is checked even if it is currently showing light mode. in short in compact view, menu option is correctly checked even if theme is displayed incorrectly.
Makes sense. But let's first merge this PR and then the follow-up can fix that :)

>     3. dark mode icon seems a little pixelated
 and also moon has a light gray area on the bottom edge but I think this can be fixed as follow-up.

Yes, this is just a placeholder, there is already a person interesting in contribute with an icon. See https://github.com/CollaboraOnline/online/issues/5054#issuecomment-1199931471

**Comment by lpranam** [comment]
...and also you can put all those changes in the same commit you don't need to have 2 commits here. 
multiple ways you can do it
1. you can reset both your commits and recommit everything in one commit (to reset top 2 commits `git reset HEAD~2`)
2. you can just reset the last commit and amend the previous commit
3. you can squash them. https://gist.github.com/lpranam/4ae996b0a4bc37448dc80356efbca7fa

**Comment by pedropintosilva** [comment]
Merged to feature branch with PR #5102

**Pull request closed without merging**